.github/workflows/changelog-check.yml

```diff
@@ -0,0 +1,80 @@
+# **what?**
+# Checks that a file has been committed under the /.changes directory
+# as a new CHANGELOG entry.  Cannot check for a specific filename as
+# it is dynamically generated by change type and timestamp.
+# This workflow should not require any secrets since it runs for PRs
+# from forked repos.
+# By default, secrets are not passed to workflows running from
+# a forked repo.
+
+# **why?**
+# Ensure code change gets reflected in the CHANGELOG.
+
+# **when?**
+# This will run for all PRs going into main and *.latest.
+
+name: Check Changelog Entry
+
+on:
+  push:
+    branches:
+      - "main"
+      - "*.latest"
+  pull_request:
+  workflow_dispatch:
+
+permissions: read-all
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  changelog:
+    name: changelog
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Check out the repository (PR)
+        if: github.event_name == 'pull_request_target'
+        uses: actions/checkout@v2
+        with:
+          persist-credentials: false
+          ref: ${{ github.event.pull_request.head.sha }}
+      - name: Check if changelog file was added
+        # https://github.com/marketplace/actions/paths-changes-filter
+        # For each filter, it sets output variable named by the filter to the text:
+        #  'true' - if any of changed files matches any of filter rules
+        #  'false' - if none of changed files matches any of filter rules
+        # also, returns:
+        #  `changes` - JSON array with names of all filters matching any of the changed files
+        uses: dorny/paths-filter@v2
+        id: filter
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          filters: |
+            changelog:
+              - added: '.changes/unreleased/**.yaml'
+      - name: Check a file has been added to .changes/unreleased
+        uses: actions/github-script@v6
+        if: steps.filter.outputs.changelog == 'false'
+        with:
+          script: |
+            github.rest.issues.createComment({
+              issue_number: context.issue.number,
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              body: '👋 Please add a changelog entry!'
+            })
+      - name: Check a file has been added to .changes/unreleased
+        uses: actions/github-script@v6
+        if: steps.filter.outputs.changelog == 'true'
+        with:
+          script: |
+            github.rest.issues.createComment({
+              issue_number: context.issue.number,
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              body: '👋 Thanks for adding a changelog entry!'
+            })
```

Comment:
@leahwicz any pointers on the action would be helpful.  I got it to detect when a new yaml file is added in the expected directory (`.changes/unreleased`) but I'm having trouble getting the action to post a message in the PR letting the contributor know what to do.

Reply:
Hey Emily I'm curious if the problem is the read-all permission on the Action. It is trying to write to the PR but isn't allowed. The article below tells you how to give write permissions just to PRs but as s quick test why don't you just remove the permission scope all together (it will default to write permission I think) and see if it works

https://github.blog/changelog/2021-04-20-github-actions-control-permissions-for-github_token/